meinberlin/apps/budgeting/api.py

```diff
@@ -56,36 +56,19 @@ def list(self, request, *args, **kwargs):
         filters = {}
 
         # category filter
-        categories = Category.objects.filter(
-            module=self.module
-        )
-        if categories:
-            category_choices = [('', _('All')), ]
-            if has_icons(self.module):
-                category_icons = []
-            for category in categories:
-                category_choices += (str(category.pk), category.name),
-                if has_icons(self.module):
-                    icon_name = getattr(category, 'icon', None)
-                    icon_url = get_category_icon_url(icon_name)
-                    category_icons += (str(category.pk), icon_url),
-
+        category_choices, category_icons = \
+            self.get_category_choices_and_icons()
+        if category_choices:
```

Comment:
needed to split this into different functions as linter told me this has gotten too complex :see_no_evil: